migrations/2025-04-25-090000_trusted-publishing/up.sql

```diff
@@ -0,0 +1,61 @@
+create table trustpub_configs_github
+(
+    id serial primary key,
+    created_at timestamptz not null default now(),
+    crate_id int not null references crates on delete cascade,
+    repository_owner varchar not null,
+    repository_owner_id int not null,
+    repository_name varchar not null,
```

Comment:
Should we also be tracking repository ID?

Reply:
we use the owner ID to protect against resurrection attacks, but I'm not sure if the same would apply for the repository ID. if the user recreates the repository with the same name, do we really want to break the publishing workflow because of it?

aside from that, we might not be able to get the repository ID for private repositories, unless we use the auth token of a repository owner. that might work right now with GitHub being our only auth provider, but would cause some issues if that were to change eventually.

Reply:
Just chiming in from PyPI's side: the reasons @Turbo87 are the same ones we don't track the repository ID either -- in the best case it's no stronger than the resurrection guarantee of the owner ID, but it's hard/annoying to obtain in the private repo case 🙂